core/primitives/src/optimistic_block.rs

```diff
@@ -0,0 +1,82 @@
+use borsh::{BorshDeserialize, BorshSerialize};
+use near_crypto::Signature;
+
+use crate::block::BlockHeader;
+use crate::hash::{hash, CryptoHash};
+use crate::merkle::combine_hash;
+use crate::stateless_validation::SignatureDifferentiator;
+use crate::types::BlockHeight;
+
+#[derive(BorshSerialize, BorshDeserialize, Clone, Debug, Eq, PartialEq)]
```

Comment:
Also it should derive ProtocolSchema: https://github.com/near/nearcore/blob/master/docs/architecture/how/serialization.md#protocolschema